binary_search_tree/tree.py

```diff
@@ -1,59 +1,148 @@
 class TreeNode:
-    def __init__(self, key, val = None):
-        if val == None:
-            val = key
+    def __init__(self, key, value = None):
+        if value == None:
+            value = key
 
         self.key = key
-        self.value = val
+        self.value = value
         self.left = None
         self.right = None
         
-
-
 class Tree:
     def __init__(self):
         self.root = None
 
-    # Time Complexity: 
-    # Space Complexity: 
+    # Time Complexity: O(log n) if balanced
+    # Space Complexity: O(1)
     def add(self, key, value = None):
```

Comment:
👍 However you made a recursive solution which means your space complexity is O(log n).  This is due to the recursive call stack.